### PR TITLE
Add changelog for v0.5.0.

### PR DESCRIPTION
**Pull request by ehpor**
Add changelog for v0.5.0.

### CODE REVIEW

File: doc/changelog.rst [hunk 1]
```diff
@@ -1,6 +1,35 @@
...
+* Autodetect FFTs when doing general Fourier transforms (`@ehpor <https://github.com/ehpor>`__)
+* Added file IO for Field, Grid and ModeBasis (`@ehpor <https://github.com/ehpor>`__)
+* Allow the use of 32bit floating point Wavefronts (`@ehpor <https://github.com/ehpor>`__)
+* Added the LUVOIR-B, GMT, TMT, Hale, HabEx, Hubble, JWST and VLTI telescope apertures (`@ivalaginja <https://github.com/ivalaginja>`__, `@syhaffert <https://github.com/syhaffert>`__ & `@ehpor <https://github.com/ehpor>`__)
+* Improved the speed of spiders and segmented apertures by 4x (`@ehpor <https://github.com/ehpor>`__)
+* Knife edge coronagraph now works in all four directions (`@syhaffert <https://github.com/syhaffert>`__ & `@ehpor <https://github.com/ehpor>`__))
+* Improved support for animations (`@ehpor <https://github.com/ehpor>`__)
+* Added focal-length parameter to Lyot coronagraph (`@syhaffert <https://github.com/syhaffert>`__)
+* Added a telescope pupil introductory tutorial (`@ehpor <https://github.com/ehpor>`__)
+* Allow single-mode fibers to be put at arbitrary positions (`@syhaffert <https://github.com/syhaffert>`__)
```
Comment (syhaffert) [inline doc/changelog.rst:22]: step-index fibers (can have any amount of modes).

File: doc/changelog.rst [hunk 2]
```diff
@@ -1,6 +1,35 @@
...
+This HCIPy release provides further speed improvements for Fourier transforms, among some other minor improvements and bugfixes. See the list of all changes below for a summary. This release can now also be installed with Conda. We dropped support for Python 3.6. This version supports Python 3.7+.
+
+List of major changes
+~~~~~~~~~~~~~~~~~~~~~
+
+* Further speed improvements using NumExpr and in-place operations (`@ehpor <https://github.com/ehpor>`__).
+* Autodetect FFTs when doing general Fourier transforms (`@ehpor <https://github.com/ehpor>`__)
+* Added file IO for Field, Grid and ModeBasis (`@ehpor <https://github.com/ehpor>`__)
+* Allow the use of 32bit floating point Wavefronts (`@ehpor <https://github.com/ehpor>`__)
+* Added the LUVOIR-B, GMT, TMT, Hale, HabEx, Hubble, JWST and VLTI telescope apertures (`@ivalaginja <https://github.com/ivalaginja>`__, `@syhaffert <https://github.com/syhaffert>`__ & `@ehpor <https://github.com/ehpor>`__)
```
Comment (syhaffert) [inline doc/changelog.rst:16]: let's leave the VLTI for the next release. I think the ELT was also added in this release.

Comment (ehpor) [reply doc/changelog.rst]: Agreed about VLTI. I might have missed ELT. Let me check.

File: doc/changelog.rst (hunk 1 above)
Comment (ehpor) [reply doc/changelog.rst]: Fixed.

File: doc/changelog.rst (hunk 2 above)
Comment (ehpor) [reply doc/changelog.rst]: Yep, I missed the ELT. I don't know how, since it's extremely large. Fixed.

**Review by syhaffert — DISMISSED** [review]

File: doc/changelog.rst (hunk 2 above)
Comment (syhaffert) [reply doc/changelog.rst]: Yeah it was part of my major PR that included multiple things.

Comment (ehpor) [reply doc/changelog.rst]: Makes sense. I mostly looked at the names of merged PRs.

Comment (ehpor) [reply doc/changelog.rst]: One more reason to automate the writing of release notes, tbh.

**Comment by ehpor** [comment]
Rebased and added release date.

Ran all tests (also long-running) locally and everything passed. Built documentation without (relevant) errors/warnings.

**Review by syhaffert — APPROVED** [review]
Time for merging.

**Pull request merged**